src/Contexts/Adaptors/EVMAdaptors.tsx

```diff
@@ -301,19 +300,30 @@ export const EVMHomeAdaptorProvider = ({
       }${decodedRecipient.substr(2)}`; // recipientAddress (?? bytes)
 
       try {
+        const lastBlock = await provider?.getBlock(-1);
```

Comment:
Since this logic is reused in 3 different places, can we bundle it in one common function?

Reply:
Yeah I abstracted it. 👍